packages/ripple-keypairs/src/secp256k1.ts

```diff
@@ -59,19 +58,18 @@ export function derivePrivateKey(
     // As returned by validation_create for a given seed
     return privateGen
   }
-  const publicGen = secp256k1.g.mul(privateGen)
+  const publicGen =
+    secp256k1.ProjectivePoint.BASE.multiply(privateGen).toRawBytes(true)
   // A seed can generate many keypairs as a function of the seed and a uint32.
   // Almost everyone just uses the first account, `0`.
   const accountIndex = opts.accountIndex || 0
-  return deriveScalar(publicGen.encodeCompressed(), accountIndex)
-    .add(privateGen)
-    .mod(order)
+  return (deriveScalar(publicGen, accountIndex) + privateGen) % order
```

Comment:
TODO: check if use of % is ok, else use @noble/curves/abstract/mod

Reply:
In Niq speak: hasher.first256BN(), which uses bytesToNumberBE(), will always return a non-negative integer, and this propagates up the calculation. Thus direct use of the `%` is appropriate.

In ChatGPT speak:

Yes, it is safe to use return (deriveScalar(publicGen, accountIndex) + privateGen) % order in this context because all the components involved are non-negative integers.

deriveScalar(publicGen, accountIndex) generates a scalar key that is less than the order of the curve secp256k1.CURVE.n (as ensured by the deriveScalar function).
privateGen is also a scalar key derived similarly, so it is also less than secp256k1.CURVE.n.
Adding these two together (deriveScalar(publicGen, accountIndex) + privateGen) will result in a number that is less than 2 * secp256k1.CURVE.n.
Performing modulus operation (% order) will bring this value back within the order of the curve, maintaining the properties required for operating within the curve's finite field.
In the context of elliptic curve cryptography, all operations are performed within the finite field defined by the curve's order. Any calculation result should be within the range from 0 to order-1 inclusive. By doing modulus operation, we ensure that the private key is within the correct range.

So, to sum up, using the operation return (deriveScalar(publicGen, accountIndex) + privateGen) % order is correct and safe in this context, assuming hasher.first256BN() returns a non-negative number.